go-controller/pkg/testing/mock_lrp.go

```diff
@@ -0,0 +1,61 @@
+package testing
+
+import (
+	"fmt"
+
+	goovn "github.com/ebay/go-ovn"
+	"k8s.io/klog/v2"
+)
+
+// Add logical port to router
+func (mock *MockOVNClient) LRPAdd(lr string, lrp string, mac string, network []string, peer string, external_ids map[string]string) (*goovn.OvnCommand, error) {
+	klog.V(5).Infof("Adding lrp %s to router %s", lrp, lr)
+	extIdsMap := make(map[interface{}]interface{})
+	for k, v := range external_ids {
+		extIdsMap[k] = v
+	}
+	return &goovn.OvnCommand{
+		Exe: &MockExecution{
+			handler: mock,
+			op:      OpAdd,
+			table:   LogicalRouterPortType,
+			objName: lrp,
+			obj:     &goovn.LogicalRouterPort{Name: lrp, UUID: FakeUUID, MAC: mac, Networks: network, Peer: peer, ExternalID: extIdsMap},
+		},
+	}, nil
+}
+
+// Delete lrp from its attached router
+func (mock *MockOVNClient) LRPDel(lr string, lrp string) (*goovn.OvnCommand, error) {
+	klog.V(5).Infof("Deleting lrp %s", lrp)
+	return &goovn.OvnCommand{
+		Exe: &MockExecution{
+			handler: mock,
+			op:      OpDelete,
+			table:   LogicalRouterPortType,
+			objName: lrp,
+		},
+	}, nil
+}
+
+// Get all lrport by lrouter
+func (mock *MockOVNClient) LRPList(lr string) ([]*goovn.LogicalRouterPort, error) {
+	klog.V(5).Infof("LRPList called for lr: %s", lr)
```

Comment:
@palanik1 I'm not quite sure how to solve it, but because LRPAdd() isn't saving *which* logical router (`lr`) the port being added is for, LRPList() won't be able to filter the logical router port cache for only ports owned by the requested router.

I wonder how this is solved for other OVN types? LSPs don't have an LSPList method.

One way to solve it would be to embed the goovn.LRP struct into another that we use in the cache, like so:

```
type MockLogicalRouterPort struct {
    goovn.LogicalRouterPort
    lr string
}
```
and keep that as the *internal* type in the cache, but return only the embedded LogicalRouterPort struct for the LRPList() array.